### PR TITLE
Removing the deprecated upload-scripts feature

### LINKED ISSUE #9865: Remove script upload through admin endpoints
### Description

Ability to upload JavaScript through the admin endpoints was a huge vulnerability, and has been deprecated for a long time now, and should be removed completely.

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #9865 

- [x] Removes the `upload-script` feature from the list of features
- [x] Update the base test suite to deploy scripts and use them in tests that rely on running scripts (policies, mappers, and authenticators)
- [x] Remove script-based providers from the list of available providers
- [x] Update the console tests to not rely on managing script providers through the admin console
- [x] Remove the JS editor from the admin console
- [x] Replace JS editor with a text area to display JS code in policies, mappers, and authenticators
- [x] Update documentation
- [x] Update the release notes
- [x] Update the migration guide

### CODE REVIEW

**Comment by pedroigor** [comment]
Documentation changes https://github.com/keycloak/keycloak-documentation/pull/1462.

**Review by stianst — DISMISSED** [review]
LGTM

File: services/src/main/java/org/keycloak/authentication/authenticators/browser/ScriptBasedAuthenticatorFactory.java [hunk 1]
```diff
@@ -153,6 +142,6 @@ public List<ProviderConfigProperty> getConfigProperties() {
 
     @Override
     public boolean isSupported() {
-        return Profile.isFeatureEnabled(Profile.Feature.SCRIPTS) && Profile.isFeatureEnabled(Profile.Feature.UPLOAD_SCRIPTS);
+        return Profile.isFeatureEnabled(Profile.Feature.SCRIPTS);
```
Comment (mposolda) [inline services/src/main/java/org/keycloak/authentication/authenticators/browser/ScriptBasedAuthenticatorFactory.java:145]: @stianst @pedroigor AFAIK this means that rest-based provider `ScriptBasedAuthenticator` will be now enabled when "scripts" feature is enabled. Is it really expected behaviour? Should not this method always return `false` now to make sure provider is not available anywhere?

Note that I've did just very quick scan of this PR and I might be missing some details. Just wanted to doublecheck if this is expected or rather a mistake...

File: services/src/main/java/org/keycloak/protocol/oidc/mappers/ScriptBasedOIDCProtocolMapper.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ public String getHelpText() {
 
   @Override
   public boolean isSupported() {
-    return Profile.isFeatureEnabled(Profile.Feature.SCRIPTS) && Profile.isFeatureEnabled(Profile.Feature.UPLOAD_SCRIPTS);
+    return Profile.isFeatureEnabled(Profile.Feature.SCRIPTS);
```
Comment (mposolda) [inline services/src/main/java/org/keycloak/protocol/oidc/mappers/ScriptBasedOIDCProtocolMapper.java:123]: Same question like for `ScriptBasedAuthenticator` .

**Review by mposolda — CHANGES_REQUESTED** [review]
@stianst @pedroigor I have some questions inline regarding this PR. I am temporarily marking this as "Request changes" until these are explained.

File: services/src/main/java/org/keycloak/authentication/authenticators/browser/ScriptBasedAuthenticatorFactory.java (hunk 1 above)
Comment (pedroigor) [reply services/src/main/java/org/keycloak/authentication/authenticators/browser/ScriptBasedAuthenticatorFactory.java:145]: The provider is no longer registered as a service, that is why I did not change here.

File: services/src/main/java/org/keycloak/protocol/oidc/mappers/ScriptBasedOIDCProtocolMapper.java (hunk 2 above)
Comment (pedroigor) [reply services/src/main/java/org/keycloak/protocol/oidc/mappers/ScriptBasedOIDCProtocolMapper.java:123]: The provider is no longer registered as a service, that is why I did not change here.

File: services/src/main/java/org/keycloak/authentication/authenticators/browser/ScriptBasedAuthenticatorFactory.java (hunk 1 above)
Comment (mposolda) [reply services/src/main/java/org/keycloak/authentication/authenticators/browser/ScriptBasedAuthenticatorFactory.java:145]: Yes, true. Thanks for pointing this!

File: services/src/main/java/org/keycloak/protocol/oidc/mappers/ScriptBasedOIDCProtocolMapper.java (hunk 2 above)
Comment (mposolda) [reply services/src/main/java/org/keycloak/protocol/oidc/mappers/ScriptBasedOIDCProtocolMapper.java:123]: Yes, true. Thanks for pointing this!

**Review by mposolda — COMMENTED** [review]
Thanks for the explanation of the concerns above.

File: services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java [hunk 3]
```diff
@@ -497,6 +499,19 @@ public Response addExecutionToFlow(@PathParam("flowAlias") String flowAlias, Map
...
+        if (f instanceof ConfiguredProvider) {
+            ConfiguredProvider internalProviderFactory = (ConfiguredProvider) f;
+            AuthenticatorConfigModel config = internalProviderFactory.getConfig();
+
+            if (config != null) {
+                // creates a default configuration if the factory defines one
+                // useful for internal providers that already provide a built-in configuration
+                AuthenticatorConfigRepresentation configRepresentation = ModelToRepresentation.toRepresentation(
+                        config);
+                newExecutionConfig(execution.getId(), configRepresentation);
```
Comment (mhajas) [inline services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java:511]: Should we close the response created by this method call?

**Review by mhajas — COMMENTED** [review]
Thanks for the changes @pedroigor. I added two minor comments to the code.

Also, I think we should run the pipeline for this as I am afraid this change can break some tests in the remote mode. Let me know if you want me to trigger it.

File: services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java (hunk 3 above)
Comment (pedroigor) [reply services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java:511]: Sorry, which response should we close?

File: authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/js/DeployedScriptPolicyFactory.java [hunk 4]
```diff
@@ -66,7 +66,9 @@ public JSPolicyRepresentation toRepresentation(Policy policy, AuthorizationProvi
 
         representation.setId(policy.getId());
         representation.setName(policy.getName());
-        representation.setDescription(metadata.getDescription());
+        if (policy.getDescription() == null) {
+            representation.setDescription(metadata.getDescription());
+        }
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/js/DeployedScriptPolicyFactory.java:71]: The description is set elsewhere. See https://github.com/keycloak/keycloak/blob/99c06d11023689875b48ef56442c90bdb744c869/server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java#L950.

**Comment by pedroigor** [comment]
@mhajas Thanks, I appreciate if you can start a job and see how it goes.

File: services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java (hunk 3 above)
Comment (pedroigor) [reply services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java:511]: Ahh right. Will close it.

**Review by stianst — DISMISSED** [review]

**Review by mposolda — DISMISSED** [review]
I am approving. It will be good to rely on the review from @mhajas who probably deeper review of this.

Also I am approving as long as it is ok to have "remote" tests failures (leaving to QA team if it is fine or not)

**Comment by pedroigor** [comment]
@stianst @mposolda We should exclude the tests failing in the pipeline for the remote server.

I can do it now or as a follow up if you want to merge the PR as is.

**Comment by mhajas** [comment]
@pedroigor I was able to fix the tests another way than excluding the tests, could you have a look at this commit: https://github.com/mhajas/keycloak/commit/a38d894af60984a58d9d31e53215add53f02e57c

**Review by mhajas — DISMISSED** [review]
This PR: https://github.com/keycloak/keycloak/pull/11534 should be merged right after this one to fix remote tests.

**Review by mhajas — APPROVED** [review]
Thanks @pedroigor

**Review by stianst — APPROVED** [review]

**Pull request merged**